terraform/provision/ingress.tf

```diff
@@ -48,25 +33,38 @@ resource "helm_release" "ingress_nginx" {
   name       = "ingress-nginx"
   chart      = "ingress-nginx"
   repository = "https://kubernetes.github.io/ingress-nginx"
-  # version    = "0.5.2"
 
   namespace       = "kube-system"
   cleanup_on_fail = "true"
   atomic          = "true"
   timeout         = 600
 
   dynamic "set" {
-    for_each = local.ingress_gateway_annotations
+    for_each = {
```

Comment:
@srinirei these are the changes I thought you might want to check out... In consolidating all the settings for the ingress-nginx Helm chart here, I took quotes off of the HCL values and removed the "type=string" forcing in the resulting content. This resolved a problem for me (where the string "false" was being interpreted in the chart as "true") and I thought it may inadvertently help you as well.